### PR TITLE
Add bs-table

### PR DESCRIPTION
**Pull request by luisvt**

### CODE REVIEW

File: lib/components/table/column_directive.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  /// * 'NO_SORTABLE': for column not sortable
+  @Input() String sort;
+
+  /// Name of the field associated with the column
+  @Input() String fieldName;
+
+  /// Value displayed in the column header
+  @Input() String header;
+
+
```
Comment (Pacane) [inline lib/components/table/column_directive.dart:22]: rm empty line

File: lib/components/table/column_directive.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  BsColumnDirective(this.tableComponent);
+
+  BsTableComponent tableComponent;
+
+  @override
+  ngOnInit() {
+    tableComponent.columns.add(this);
+  }
+}
```
Comment (Pacane) [inline lib/components/table/column_directive.dart:31]: you should dartfmt

File: lib/components/table/table_component.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    pageNumberChange.emit(_pageNumber);
+  }
+
+  @Output() EventEmitter<num> pageNumberChange = new EventEmitter();
+
+  @Output() EventEmitter<num> totalItemsChange = new EventEmitter();
+
+  @HostListener('pageNumberChange')
+  void updatePage() {
+    var start = (pageNumber - 1) * itemsPerPage;
```
Comment (Pacane) [inline lib/components/table/table_component.dart]: could be called `startIndex` and `endIndex`

File: lib/components/table/table_directives.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
+library table_directives;
+
+import 'dart:html';
+
+import 'dart:math';
+import "package:angular2/core.dart";
+
+part 'column_directive.dart';
+part 'table_component.dart';
+
```
Comment (Pacane) [inline lib/components/table/table_directives.dart:9]: formatting

File: web/components/table/table_demo.dart [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import 'package:ng_bootstrap/components/table/table_directives.dart';
+import 'package:node_shims/js.dart';
+import 'table_data.dart';
+
+// webpack html imports
+@Component (selector: 'table-demo',
+    templateUrl: 'table_demo.html',
+    directives: const [NG_BOOTSTRAP_TABLE_DIRECTIVES, Pagination])
+class TableDemoComponent implements OnInit {
+  List rows = [];
```
Comment (Pacane) [inline web/components/table/table_demo.dart:11]: formatting?

File: web/components/table/table_demo.dart [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import 'dart:math';
+import 'package:angular2/core.dart';
+import 'package:ng_bootstrap/components/pagination/pagination.dart';
+import 'package:ng_bootstrap/components/table/table_directives.dart';
+import 'package:node_shims/js.dart';
+import 'table_data.dart';
+
+// webpack html imports
```
Comment (Pacane) [inline web/components/table/table_demo.dart]: what's this comment for?

File: web/components/table/table_demo.dart [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  void ngOnInit() {
+    this.filterRows();
+  }
+
+  void filterRows() {
+    if (falsey(config['filtering'])) {
+      rows = data.toList();
+    } else {}
```
Comment (Pacane) [inline web/components/table/table_demo.dart]: empty else clause. Is that really what you want?

Comment (Pacane) [inline web/components/table/table_demo.dart]: I suspect you want the row initialization below in to be in the else?

File: web/components/table/table_demo.dart (hunk 5 above)
Comment (luisvt) [inline web/components/table/table_demo.dart]: ups that's from original TS files

File: web/components/table/table_demo.dart (hunk 7 above)
Comment (luisvt) [inline web/components/table/table_demo.dart]: I think most likely I wrongly close that else, I'm going to correct it

File: web/components/table/table_demo.dart (hunk 5 above)
Comment (luisvt) [inline web/components/table/table_demo.dart:11]: it's already formatted

File: lib/components/table/table_directives.dart (hunk 4 above)
Comment (luisvt) [inline lib/components/table/table_directives.dart:9]: it's already formatted

**Pull request merged**